CMS-Enterprise/sbom-harbor
# feat(gh_provider): GitHub Provider Integration

## Summary

GitHub Provider has been integrated with the rest of the code

### Added

GitHub Provider functionality

## How to test

1. Run provider: ./target/debug/harbor ingest --provider github --debug
2. Verify that data is in Mongo
3. More to come

NOTE: Tests _were_ failing because there is no mongo instance for the tests to connect to.  I thought we had a sidecar, but it looks like this will need to be a manual test too? 

ODDITY: I _had_ a clippy warning that is killing the build:
```
error: explicit call to `.into_iter()` in function argument accepting `IntoIterator`
  --> sdk/core/src/services/snyk/service.rs:73:37
   |
73 |                     projects.extend(p.into_iter());
   |                                     ^^^^^^^^^^^^^ help: consider removing the `.into_iter()`: `p`
   |
note: this parameter accepts any `IntoIterator`, so you don't need to call `.into_iter()`
  --> /rustc/839e9a6e1210934fd24b15548b811a97c77138fc/library/core/src/iter/traits/collect.rs:377:18
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#useless_conversion
   = note: `-D clippy::useless-conversion` implied by `-D warnings`

error: could not compile `harbcore` (lib) due to previous error
Error: Process completed with exit code 101.
```
Yet, behold:

![image](https://github.com/CMS-Enterprise/sbom-harbor/assets/6561793/a9ee0968-3824-49d8-99d4-229121e10b11)

Line # 73 isn't changed and contains the same code.  I cannot get clippy to find the same issue locally so far.
